server/chat/backend/agent/orchestrator/synthesis.py

```
@@ -0,0 +1,286 @@
+        # existing chat pipeline handles.
+        llm = create_chat_model(model=ModelConfig.MAIN_MODEL, streaming=False)
+        structured = llm.with_structured_output(SynthesisDecision)
+
+        # Pass available role names to constrain follow-up dispatch — prevents
+        # the LLM from hallucinating role names that aren't in the registry.
+        from chat.backend.agent.orchestrator.role_registry import RoleRegistry
+        try:
+            available_roles = RoleRegistry.get_instance().list_available_roles(
+                getattr(state, "user_id", "") or ""
+            )
+        except Exception:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)